README.adoc

```diff
@@ -429,6 +429,61 @@ See https://eksctl.io/usage/vpc-networking/ for more information on the networki
 The `setup_f5_eks.sh` script exposes the Fusion proxy service on an external IP over HTTP. This is done for demo or getting started purposes. However, you're strongly encouraged to configure a K8s Ingress with TLS termination in front of the proxy service.
 See: https://aws.amazon.com/premiumsupport/knowledge-center/terminate-https-traffic-eks-acm/
 
+Our EKS script creates a classic ELB for exposing fusion proxy service. In case you need to change this behavior or create another type of ELB, you can follow these steps:
+
+. Using the AWS console for creating an https://docs.aws.amazon.com/elasticloadbalancing/latest/application/introduction.html[Application Load Balancer^]:
+
+.. Check which port is kubernetes using to allow communication, you can check this listing the fusion services: `kubectl get svc -n <namespace>`.
+.. Create an Application Load Balancer that allows communication from port 80/443. You must include your nodes as the target group and the health checker should point to the port kubernetes is using.
+.. From here we have 2 options:
+... Change the proxy service type from LoadBalancer to Nodeport using the kubernetes port.
+... Another option is deploying a nginx ingress and change the proxy service type to ClusterIP.
+
+. Using https://github.com/kubernetes-sigs/aws-alb-ingress-controller[AWS ALB Ingress Controller^]:
+.. Attach the following this https://raw.githubusercontent.com/kubernetes-sigs/aws-alb-ingress-controller/v1.1.5/docs/examples/iam-policy.json[policy^] to the role which is associated to your eks nodes.
+.. Add the helm incubator repository:
+```
+helm repo add incubator http://storage.googleapis.com/kubernetes-charts-incubator
+```
+.. Install the alb ingress controller:
+```
+helm install incubator/aws-alb-ingress-controller --set autoDiscoverAwsRegion=true --set autoDiscoverAwsVpcID=true --set clusterName=<cluster_name>
+```
+.. Change the proxy service type from LoadBalancer to Nodeport. One option is updating the following values in your fusion values yaml file and apply the changes using the upgrade script::
+```
+api-gateway:
+  service:
+    type: "NodePort"
+    port: 6764
+    targetPort: 6764
+```
+.. Create an ingress-controller for redirecting the port to the proxy service, you can use this example:
+```
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: "nginx-ingress"
+  namespace: "<your_namespace>"
+  annotations:
+    kubernetes.io/ingress.class: alb
+    alb.ingress.kubernetes.io/scheme: internet-facing
+  labels:
+    app: nginx-ingress
+spec:
+  rules:
+    - host: <your_dns_name>
+      http:
+        paths:
+          - path: /*
+            backend:
+              serviceName: "proxy"
+              servicePort: 6764
```

Comment:
Can we have this example as the values that you would have to put in your values file for the ingress to work with the `alb` ingress controller, so everything is in the same helm releases, instead of having it as a separate resource that is managed outside of helm?

Reply:
I have added the alb ingress options to the setup_eks script in the same way we configure it for gke.

Reply:
In the setup_gke script we don't create the Ingress as a separate resource, we set the values in the fusion chart correctly when we do the upgrade here: https://github.com/lucidworks/fusion-cloud-native/blob/3b4af5760a2c2cb313e2b69b7b633d3d409ac59c/setup_f5_gke.sh#L489. The chart is then responsible for creating this Ingress and so it will get cleaned up when the chart is deleted etc. Can we just document that you should add values similar to these into your values file to enable the alb ingress?